source/utilities.cc

```diff
@@ -206,22 +231,40 @@ namespace aspect
                                            "One example of where to check this is if "
                                            "Compositional fields are used, "
                                            "then check the id list "
-                                           "from `set Names of fields' in the"
+                                           "from `set Names of fields' in the "
                                            "Compositional fields subsection. "
                                            "Alternatively, if `set Names of fields' "
                                            "is not set, the default names are "
                                            "C_1, C_2, ..., C_n."));
 
-                  return_values[field_index] = Utilities::string_to_double(key_and_value[1]);
+                  const std::vector<std::string> subfield_entries = dealii::Utilities::split_string_list(key_and_value[1], '|');
+
+                  for (const auto &entry : subfield_entries)
+                    {
+                      return_map[field_index].push_back(Utilities::string_to_double(entry));
+
+                      if (subentry_structure)
+                        ++(*subentry_structure)[field_index];
+                    }
                 }
             }
+
+          for (const auto &entry: return_map)
+            for (const auto &sub_entry: entry)
+              return_values.push_back(sub_entry);
         }
       else if (Patterns::List(Patterns::Double(),1,n_fields).match(input_string))
```

Comment:
We should make sure that `allow_subentries` is false in this case.

Reply:
I would not want to do that. Consider the case that somebody wants to allow multiple values per key, but does not know the input structure the user puts into the .prm file. You would still want to allow a simple list of only a single value per key. The parameter is `allow_multiple_values_per_key`, not `enforce_multiple_values_per_key`. But you are right, I should document that this format expects exactly one value per key (and we already enforce that, because `possibly_extend_from_1_to_N` expects either 1 entry, or `n_fields` entries).